Documentation/ceph-csi-snapshot.md

```diff
@@ -99,3 +100,78 @@ kubectl delete -f cluster/examples/kubernetes/ceph/csi/rbd/pvc-restore.yaml
+name of the secret created for the cephfsplugin and `pool` to reflect the Ceph pool name.
+
+Update the value of the `clusterID` field to match the namespace that Rook is
+running in. When Ceph CSI is deployed by Rook, the operator will automatically
+maintain a configmap whose contents will match this key. By default this is
+"rook-ceph".
+
+```console
+kubectl create -f cluster/examples/kubernetes/ceph/csi/rbd/snapshotclass.yaml
+```
+
+### Volumesnapshot
+
+In [snapshot](https://github.com/rook/rook/tree/{{ branchName }}/cluster/examples/kubernetes/ceph/csi/cephfs/snapshot.yaml),
+`volumeSnapshotClassName` should be the name of the `VolumeSnapshotClass`
+previously created. The `persistentVolumeClaimName` should be the name of the
+PVC which is already created by CephFS CSI driver.
+
+```console
+kubectl create -f cluster/examples/kubernetes/ceph/csi/cephfs/snapshot.yaml
+```
+
+### Verify CephFS Snapshot Creation
+
+```console
+kubectl get volumesnapshotclass
+NAME                        DRIVER                          DELETIONPOLICY   AGE
+csi-cephfslugin-snapclass   rook-ceph.cephfs.csi.ceph.com   Delete           3h55m
+
+kubectl get volumesnapshot
+NAME                  READYTOUSE   SOURCEPVC   SOURCESNAPSHOTCONTENT  RESTORESIZE   SNAPSHOTCLASS                SNAPSHOTCONTENT                                   CREATIONTIME   AGE
+cephfs-pvc-snapshot   true         cephfs-pvc                         1Gi           csi-cephfsplugin-snapclass   snapcontent-34476204-a14a-4d59-bfbc-2bbba695652c  3h50m          3h51m
+```
+
+The snapshot will be ready to restore to a new PVC when `READYTOUSE` field of the
+`volumesnapshot` is set to true.
+
+### Restore the snapshot to a new PVC
+
+In
+[pvc-restore](https://github.com/rook/rook/tree/{{ branchName }}/cluster/examples/kubernetes/ceph/csi/cephfs/pvc-restore.yaml),
+`dataSource` should be the name of the `VolumeSnapshot` previously
+created. The `dataSource` kind should be the `VolumeSnapshot`.
+
+Create a new PVC from the snapshot
+
+```console
+kubectl create -f cluster/examples/kubernetes/ceph/csi/cephfs/pvc-restore.yaml
+```
+
+### Verify CephFS Restore PVC Creation
+
+```console
+kubectl get pvc
+NAME                 STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS      AGE
+cephfs-pvc           Bound    pvc-74734901-577a-11e9-b34f-525400581048   1Gi        RWX            rook-cephfs       55m
+cephfs-pvc-restore   Bound    pvc-95308c75-6c93-4928-a551-6b5137192209   1Gi        RWX            rook-cephfs       34s
+```
+
+## CephFS snapshot resource Cleanup
+
+To clean your cluster of the resources created by this example, run the following:
+
+```console
+kubectl delete -f cluster/examples/kubernetes/ceph/csi/cephfs/pvc-restore.yaml
+kubectl delete -f cluster/examples/kubernetes/ceph/csi/cephfs/snapshot.yaml
+kubectl delete -f cluster/examples/kubernetes/ceph/csi/cephfs/snapshotclass.yaml
```

Comment:
please add the limitations for cephfs snapshots to this doc so that people will be aware of it
* Limit of 400 snapshots per filesystem
* PVC cannot be deleted if it's having snapshots